src/transformers/generation_beam_search.py

```diff
@@ -0,0 +1,356 @@
+
+
+PROCESS_INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size * num_beams, sequence_length)`):
+            Indices of input sequence tokens in the vocabulary.
+
+            Indices can be obtained using :class:`~transformers.BertTokenizer`. See
+            :meth:`transformers.PreTrainedTokenizer.encode` and :meth:`transformers.PreTrainedTokenizer.__call__` for
+            details.
+
+            `What are input IDs? <../glossary.html#input-ids>`__
+        next_scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, 2 * num_beams)`):
+            Current scores of the top :obj:`2 * num_beams` non-finished beam hypotheses.
+        next_tokens (:obj:`torch.LongTensor` of shape :obj:`(batch_size, 2 * num_beams)`):
+            :obj:`input_ids` of the tokens corresponding to the top :obj:`2 * num_beams` non-finished beam hypotheses.
+        next_indices (:obj:`torch.LongTensor` of shape :obj:`(batch_size, 2 * num_beams)`):
+            beam indices indicating to which beam hypothesis the :obj:`next_tokens` correspond.
+        kwargs:
+            Additional kwargs including special tokens, such as :obj:`eos_token_id` and :obj:`pad_token_id`.
+
+    Return: :obj:`UserDict` composed of the fields as defined above:
+
+            - next_beam_scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size * num_beams)`) -- updated scores of
+              all non-finished beams.
+            - next_beam_tokens (:obj:`torch.FloatTensor` of shape :obj:`(batch_size * num_beams)`) -- next tokens to be
+              added to the non-finished beam_hypotheses.
+            - next_beam_indices (:obj:`torch.FloatTensor` of shape :obj:`(batch_size * num_beams)`) -- beam indices
+              indicating to which beam the next tokens shall be added.
+
+"""
+
+FINALIZE_INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size * num_beams, sequence_length)`):
+            Indices of input sequence tokens in the vocabulary.
+
+            Indices can be obtained using :class:`~transformers.BertTokenizer`. See
+            :meth:`transformers.PreTrainedTokenizer.encode` and :meth:`transformers.PreTrainedTokenizer.__call__` for
+            details.
+
+            `What are input IDs? <../glossary.html#input-ids>`__
+        final_beam_scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size * num_beams)`):
+            The final scores of all non-finished beams.
+        final_beam_tokens (:obj:`torch.FloatTensor` of shape :obj:`(batch_size * num_beams)`):
+            The last tokens to be added to the non-finished beam_hypotheses.
+        final_beam_indices (:obj:`torch.FloatTensor` of shape :obj:`(batch_size * num_beams)`):
+            The beam indices indicating to which beam the :obj:`final_beam_tokens` shall be added.
+        kwargs:
+            Additional kwargs including special tokens, such as :obj:`eos_token_id` and :obj:`pad_token_id`.
```

Comment:
```suggestion
            Additional keyword arguments including special tokens, such as :obj:`eos_token_id` and :obj:`pad_token_id`.
```
Same as above. Where do they go?